freshmaker/handlers/botas/botas_shipped_advisory.py

```diff
@@ -270,24 +306,107 @@ def handle(self, event):
                 to_rebuild_digests.add(digest)
 
         if not to_rebuild_digests:
-            msg = f"No bundle images to rebuild for advisory {event.advisory.name}"
+            msg = f"No bundle images to rebuild for advisory {self.event.advisory.name}"
             self.log_info(msg)
             db_event.transition(EventState.SKIPPED, msg)
             db.session.commit()
             return []
 
-        builds = self._prepare_builds(db_event, bundles_by_digest,
-                                      to_rebuild_digests)
+        bundles_to_rebuild = list(map(lambda x: bundles_by_digest[x],
+                                      to_rebuild_digests))
+        return bundles_to_rebuild
 
-        # Reset context to db_event.
-        self.set_context(db_event)
+    def _handle_manual_rebuild(self, db_event):
+        """
+        Handle manual rebuild submitted by Release Driver for an advisory created by Botas
 
-        self.start_to_build_images(builds)
-        msg = f"Advisory {db_event.search_key}: Rebuilding " \
-              f"{len(db_event.builds.all())} bundle images."
-        db_event.transition(EventState.BUILDING, msg)
+        :param db_event: database event that represent rebuild event
+        :rtype: list
+        :return: list of advisories that should be rebuilt
+        """
+        default_bundle_data = {
+            'nvr': '',
+            'update': {},
+            'append': {},
+            'pullspecs': [],
+        }
+        # map with old pullspec and bundle info
+        old_pullspec_to_bundle_map = dict()
+        # get all replaced pullspecs from provided 'bundle_images'
+        for bundle_nvr in self.event.bundle_images:
+            artifact_build = db.session.query(ArtifactBuild).filter(
+                ArtifactBuild.rebuilt_nvr == bundle_nvr,
+                ArtifactBuild.type == ArtifactType.IMAGE.value,
+            ).one_or_none()
+            pullspec_overrides = artifact_build.bundle_pullspec_overrides
+            for pullspec in pullspec_overrides:
+                old_pullspec = pullspec.get('_old', None)
+                if old_pullspec is not None:
+                    old_pullspec_to_bundle_map.setdefault(
```

Comment:
If I'm reading this PR correctly, I think the logic is reversed. It should be that for all `container_images` (these are actually bundle images but just reusing the key that the API already accepts), check if any of the pullspecs that Freshmaker replaced in each bundle image in `bundle_images` has an "_old" (as you call it in this PR) pullspec that matches a pinned pullspec in the container image (from the `container_images` list).

Does that make sense?

Reply:
Yes, so at first I am collecting all the '_old' pullspecs from the submitted `bundle_images`. And then comparing them with the pullspecs of images from `container_images` list. I have it there  
```
        for container_image_nvr in self.container_images:
            # Fetch buildinfo
            buildinfo = koji_api.get_build(container_image_nvr)
            # Get the original pullspecs
            pullspecs = (
                buildinfo.get('extra', {})
                         .get('image', {})
                         .get('operator_manifests', {})
                         .get('related_images', {})
                         .get('pullspecs', [])
            )
            for pullspec in pullspecs:
                if pullspec.get('new') in old_pullspec_to_bundle_map:
                ....
``` 
So at the end I am comparing `container_images->pullspecs->new` with the `bundle_images->pullspecs->_old` for every **pullspec** in every **container_image**. Is it right?